src/extension.ts

```diff
@@ -57,46 +136,49 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
 			return;
 		}
 		const activeDocumentUri = activeEditor.document.uri;
-		let activeWorkspaceFolder: vscode.WorkspaceFolder | undefined;
-		workspaceFolders.forEach(element => {
-			console.log(element.uri.toString());
-			console.log(activeDocumentUri.toString());
-			if (activeDocumentUri.toString().startsWith(element.uri.toString())) {
-				activeWorkspaceFolder = element;
-			}
-		});
-		if (isUndefined(activeWorkspaceFolder)) {
+
+
+		let cwd = detectRootFromHoverConfig(activeDocumentUri) ||
+			detectPubspecFromDir(path.dirname(activeDocumentUri.fsPath)) ||
+			detectRootFromWorkspaces(activeDocumentUri, workspaceFolders)
+
+		if (cwd == "") {
 			vscode.window.showErrorMessage('No active workspace found to launch Hover for.');
 			return;
 		}
 
-		let cwd = activeWorkspaceFolder.uri.fsPath;
-		let target = "lib/main_desktop.dart"; // TODO: make configurable, probably best done through launch configurations with a param...
+		let workspace = detectWorkspaceFor(cwd, workspaceFolders)
+
+		if (isUndefined(workspace)) {
+			vscode.window.showErrorMessage('No active workspace found to launch Hover for.');
+			return;
+		}
+
+		const config = configOrDefault(path.join(cwd, "go", "hover.yaml"), {target: "lib/main_desktop.dart"})
 
 		runHoverRun(
 			[
-				"--target", target,
+				"--target", config.target,
 				"--colors=false", // TODO: Fix colors in output stream, if possible.
 			],
 			cwd,
 		).then((observatoryUri: string) => {
 			console.log("runHover started, have observatoryUri");
 			vscode.window.showInformationMessage('Starting application in debug mode');
 			console.log("Attaching debugger (flutter attach)");
-			vscode.debug.startDebugging(activeWorkspaceFolder, {
+			vscode.debug.startDebugging(workspace, {
 				name: "Flutter attached to Hover",
 				type: "dart",
 				request: "attach",
 				deviceId: "flutter-tester",
 				observatoryUri: observatoryUri,
-				program: target,
+				program: config.target,
```

Comment:
I think this needs to be prefixed with the cwd path relative within the workspace?

I have a monorepo as workspace where a subdirectory contains the flutter app. "Hover: run" now correctly runs hover using that directory as cwd, but flutter attach fails, I get a popup:

> Your launch config references a program that does not exist. If you have problems launching, check the "program" field in your ".vscode/launch.json" file.

I think the program value needs to be prefixied with the path of the cwd inside the active workspace?

Reply:
that popup is a red herring? i get it to locally but everything works fine on my end.

going to need more details on the paths involved to determine whats going on for your environment.

Reply:
but you're likely right, will have to investigate a bit.

Reply:
does hover support automatic hot reload? only thing i felt was missing when i used this.

Reply:
Hot reload works by running `flutter attach` to hook onto the observatory of the running app. Hot reload (flutter attach) works when I execute "Hover: run" in a workspace that is also the root of the flutter app, but fails when it is in a subdirectory.

Reply:
I found and fixed the issue, will merge them right away